src/core/pick.ts

```diff
@@ -1,24 +1,134 @@
 import type { StateTree } from 'pinia'
+import type { Mergeable } from '~/core/types'
+
+function merge(destination: Mergeable, source: Mergeable): Mergeable {
+  const mergingArrays = Array.isArray(destination) && Array.isArray(source)
+  const mergingObjects =
```

Comment:
can you create a small `function isObject(value: unknown) {}` util for readability concern ?